### PR TITLE
Automated cherry pick of #71653: fix cni timeout

### PR DESCRIPTION
**Pull request by nikopen**
Cherry pick of #71653 on release-1.16.

#71653: fix cni timeout

For details on the cherry pick process, see the [cherry pick requests](https://git.k8s.io/community/contributors/devel/sig-release/cherry-picks.md) page.

### CODE REVIEW

**Comment by nikopen** [comment]
@kubernetes/release-engineering

**Comment by neolit123** [comment]
@nikopen 
trying to establish if this is critical for a backport.

while the original PR is of kind/bug it seems more like a kind/feature, even if it solves potential bugs, if there is no timeout.

probably best to ask in #sig-network are fine with this backport.
@kubernetes/sig-network-pr-reviews 
@kubernetes/patch-release-team

**Review by mattjmcnaughton — COMMENTED** [review]
@neolit123 I agree that there are arguments on both sides for whether this commit should be cherry-picked onto 1.16.

Fwiw, it looks like the 1.16 release time was hoping this diff would be merged for 1.16: https://github.com/kubernetes/kubernetes/pull/71653#issuecomment-528492124

Additionally, it does seem like users are feeling the impact of this bug: https://github.com/kubernetes/kubernetes/pull/71653#issuecomment-565666282

Looking forward to hearing the opinion of the sig-network folks.

**Review by cmluciano — COMMENTED** [review]
/lgtm

based on the comments of individuals being affected by this within earlier releases, I think this should be cherry-picked. There is also feedback about this affecting 1.15 releases too.

**Comment by bowei** [comment]
Ping @kubernetes/patch-release-team

**Review by freehan — APPROVED** [review]
/lgtm
/approve

**Review by saschagrunert — APPROVED** [review]
/lgtm

**Review by cpanato — APPROVED** [review]
/lgtm

**Comment by nikopen** [comment]
/hold cancel
/approve

@saschagrunert @cpanato I think you have to /approve to remove the do-not-merge?

**Comment by justaugustus** [comment]
@nikopen -- The Patch Release Team approves cherry picks, not Branch Management.
We'll be reviewing patches next week. Thanks!

**Pull request merged**
